### PR TITLE
Docs: Update backend plugin forwarding of headers

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**

Updates backend plugin documentation in regards to automatic forwarding of HTTP headers based on https://github.com/grafana/grafana-plugin-sdk-go/pull/679 changes and breaking changes explained in https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.161.0.

**Why do we need this feature?**

Plugin authors need latest information.

**Who is this feature for?**

Plugin authors.

**Special notes for your reviewer:**

Please check that:
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by xnyo — APPROVED** [review]
LGTM!

File: docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md [hunk 1]
```diff
@@ -291,14 +291,17 @@ To allow Grafana to pass the access token to the plugin, update the data source
 
 When configured, Grafana can forward authorization HTTP headers such as `Authorization` or `X-ID-Token` to a backend data source. This information is available across the `QueryData`, `CallResource` and `CheckHealth` requests.
 
-To get Grafana to forward the headers, create a HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient). This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
+To get Grafana to forward the headers, create a HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient) and set the `ForwardHTTPHeaders` option to `true`, by default it's set to `false`. This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
```
Comment (wbrowne) [inline docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md:294]:
```suggestion
To get Grafana to forward the headers, create a HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient) and set the `ForwardHTTPHeaders` option to `true` (by default it's set to `false`). This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
```

**Review by wbrowne — APPROVED** [review]
Probably better to have @josmperez do a final sign off on this, but LGTM!

File: docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md [hunk 2]
```diff
@@ -291,14 +291,17 @@ To allow Grafana to pass the access token to the plugin, update the data source
 
 When configured, Grafana can forward authorization HTTP headers such as `Authorization` or `X-ID-Token` to a backend data source. This information is available across the `QueryData`, `CallResource` and `CheckHealth` requests.
 
-To get Grafana to forward the headers, create a HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient). This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
+To get Grafana to forward the headers, create a HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient) and set the `ForwardHTTPHeaders` option to `true` (by default it's set to `false`). This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
```
Comment (josmperez) [inline docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md:294]:
```suggestion
To get Grafana to forward the headers, create a HTTP client using the [Grafana plugin SDK for Go](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/httpclient) and set the `ForwardHTTPHeaders` option to `true` (by default, it's set to `false`). This package exposes request information which can be subsequently forwarded downstream and/or used directly within the plugin.
```

**Review by josmperez — CHANGES_REQUESTED** [review]
Minor style nit. LGTM.

**Review by josmperez — APPROVED** [review]

**Pull request merged**